js/ui/file_manager/file_provider/ajax.js

```diff
@@ -1,13 +1,45 @@
 import ajax from "../../../core/utils/ajax";
 import { Deferred } from "../../../core/utils/deferred";
+import { extend } from "../../../core/utils/extend";
 import { FileProvider } from "./file_provider";
 import ArrayFileProvider from "./array";
 
+/**
+* @name AjaxFileProvider
+* @inherits FileProvider
+* @type object
+* @module ui/file_manager/file_provider/ajax
+* @export default
+*/
 class AjaxFileProvider extends FileProvider {
 
     constructor(options) {
-        super();
-        this._url = options.url;
+        super(options);
+        /**
+         * @name AjaxFileProviderOptions.url
+         * @type string
+         */
+        /**
+         * @name AjaxFileProviderOptions.nameExpr
+         * @type string|function
+         */
+        /**
+         * @name AjaxFileProviderOptions.isFolderExpr
+         * @type string|function
+         */
+        /**
+         * @name AjaxFileProviderOptions.sizeExpr
+         * @type string|function
+         */
+        /**
+         * @name AjaxFileProviderOptions.dateModifiedExpr
+         * @type string|function
+         */
+        /**
+         * @name AjaxFileProviderOptions.thumbnailExpr
+         * @type string|function
```

Comment:
Do these functions accept any parameters?

Reply:
Yes, we pass file item object that came from server